src/review/logic.py

```diff
@@ -50,28 +60,134 @@ def handle_review_file(_file, content_type, review_assignment, kind):
         _file._get_name().replace(',', '_').replace(';', '_')
     )
     filename = str(uuid4()) + str(path.splitext(original_filename)[1])
+    file_mime = mime.guess_type(filename)[0] or 'unknown/unknown'
 
     if content_type == 'book':
-        folder_structure = path.join(
-            settings.BASE_DIR,
-            'files',
-            'books',
-            str(review_assignment.book.id)
+        file_path = path.join(
+            settings.BOOK_DIR,
+            str(review_assignment.book.id),
+            str(filename),
         )
     else:
-        folder_structure = path.join(
-            settings.BASE_DIR,
-            'files',
-            'proposals',
-            str(review_assignment.proposal.id)
+        file_path = path.join(
+            settings.PROPOSAL_DIR,
+            str(review_assignment.proposal.id),
+            str(filename),
         )
 
-    if not path.exists(folder_structure):
-        makedirs(folder_structure)
+    with default_storage.open(file_path, 'wb') as file_stream:
+        file_stream.write(_file.read())
+
+    new_file = core_models.File(
+        mime_type=file_mime,
+        original_filename=original_filename,
+        uuid_filename=filename,
+        stage_uploaded=1,
+        kind=kind,
+        owner=review_assignment.user,
+    )
+    new_file.save()
```

Comment:
Is there a reason this isn't done in a single `core_models.File.objects.create()` statement?